app.js

```diff
@@ -1,73 +1,67 @@
-
+const { URL, DBNAME } = require('./utilities/dbSettings');
 
 
 app.use(express.json());
 app.put('/:collection', (req, res) => {
-    const payload = req.body;
-    const { collection } = req.params;
+  const payload = req.body;
+  const { collection } = req.params;
+
+  (async function mongo() {
 
-    (async function mongo() {
-        let client = new MongoClient(url, { useNewUrlParser: true });
-        try {
-            client = MongoClient.connect(url);
-            const db = (await client).db(dbName);
-            const col = await db.collection(collection.toString());
-            await col.deleteMany();
-            const response = await col.insertOne(payload);
-            res.status(201).json(response)
-        } catch (error) {
-            res.send(error);
-        }
-        (await client).close();
-    })();
+    let client = new MongoClient(URL, { useNewUrlParser: true });
+    try {
+      client = MongoClient.connect(URL);
+      const db = (await client).db(DBNAME);
+      const col = await db.collection(collection.toString());
+      await col.deleteMany();
+      const response = await col.insertOne(payload);
+      res.status(201).json(response)
+    } catch (error) {
+      res.send(error);
+    }
+    (await client).close();
+  })();
 });
 
 app.get('/:collection', (req, res) => {
-    const {collection} = req.params;
+  const { collection } = req.params;
 
-    (async function mongo() {
-        let client = new MongoClient(url, { useNewUrlParser: true });
-        try {
-            client = MongoClient.connect(url);
-            const db = (await client).db(dbName);
-            const col = await db.collection(collection.toString());
-            const text = await col.findOne()
-            res.json(text);
-        } catch (error) {
-            res.send(error);
-        }
-        (await client).close();
-    })();
+  mng.connect()
+    .then(() => mng.connObj.db.collection(collection))
+    .then(col => col.findOne())
+    .then(text => res.json(text))
+    .catch(err => {
+      console.log(err);
+      mng.close();
+    });
 });
 
 app.get('/:collection/:language', (req, res) => {
-    const { collection, language } = req.params;
+  console.log('/:collection:language');
+  const { collection, language } = req.params;
 
-    (async function mongo() {
-        let client = new MongoClient(url, { useNewUrlParser: true });
-        try {
-            client = MongoClient.connect(url);
-            const db = (await client).db(dbName);
-            const col = await db.collection(collection.toString());
-            const text = await col.findOne()
-            res.json(selectLanguage(text, language))
-        } catch (error) {
-            res.send(error);
-            console.log(error);
-        }
-        (await client).close();
-    })();
+  mng.connect()
+    .then(() => mng.connObj.db.collection(collection))
+    .then(col => col.findOne())
+    .then(text => selectLanguage(text, language))
```

Comment:
Here, because we don't handle null or undefined cases in selectlanguage, there's never a response to the browser. In my case I used a dirty hack res.send(error) but really, we should handle these cases properly.

Reply:
Yes you are right the user should know what's going on. In case of error he should be notified of that, I do agree.